lxc/query.go

```diff
@@ -109,7 +109,9 @@ func (c *cmdQuery) Run(cmd *cobra.Command, args []string) error {
 
 		// If not JSON decoding error then fail immediately.
 		if !errors.As(err, &jsonSyntaxError) && !errors.As(err, &jsonUnmarshalTypeError) {
-			return err
+			if (err.Error() != "EOF") {
```

Comment:
No need for the brackets in Go, this may also want to try `if err != io.EOF {`

Reply:
```diff
diff --git a/lxc/query.go b/lxc/query.go
index 8554e32f2..7e20fa678 100644
--- a/lxc/query.go
+++ b/lxc/query.go
@@ -108,7 +108,7 @@ func (c *cmdQuery) Run(cmd *cobra.Command, args []string) error {
                var jsonUnmarshalTypeError *json.UnmarshalTypeError
 
                // If not JSON decoding error then fail immediately.
-               if !errors.As(err, &jsonSyntaxError) && !errors.As(err, &jsonUnmarshalTypeError) {
+               if !errors.As(err, &jsonSyntaxError) && !errors.As(err, &jsonUnmarshalTypeError) && err != io.EOF {
                        return err
                }
 
```

Reply:
That should be a slightly simpler version of your fix.

Reply:
Should be fixed now!